src/_utils/focusVisibleProvider/index.tsx

```diff
@@ -17,16 +17,30 @@ const pointerMoveEventList = [
 
 const pointerDownEventList = ['mousedown', 'pointerdown', 'touchstart']
 
+export const FOCUS_VISIBLE_CSS_CLASS = 'focus-visible'
+
 type FocusVisibleProviderProps = {
   children: ReactNode
+  // Provide a global css selector for legacy components.
+  setGlobalClassName?: boolean
 }
 
 export const FocusVisibleContext = createContext(false)
 
-export const FocusVisibleProvider = ({ children }: FocusVisibleProviderProps) => {
-  /* When the provider first loads, assume the user is in pointer modality. */
+export const FocusVisibleProvider = ({
+  children,
+  setGlobalClassName,
+}: FocusVisibleProviderProps) => {
+  // When the provider first loads, assume the user is in pointer modality.
   const [hadKeyboardEvent, setHadKeyboardEvent] = useState(false)
 
+  if (setGlobalClassName) {
+    useEffect(() => {
```

Comment:
Hooks should not be called conditionally, it's [the first rule of hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level).

You should move the condition inside of the effect:

```ts
useEffect(() => {
  if (setGlobalClassName) {
    document.body.classList.toggle(FOCUS_VISIBLE_CSS_CLASS, hadKeyboardEvent)
  }
}, [hadKeyboardEvent, setGlobalClassName])
```